dist/tools/firmware_metadata/Makefile

```diff
@@ -0,0 +1,19 @@
+RIOTBASE := ../../..
+RIOT_INCLUDE = $(RIOTBASE)/sys/include
+SHA256_DIR := $(RIOTBASE)/sys/hashes
+SHA256_INCLUDE := $(RIOT_INCLUDE)/hashes
+METADATA_SRC := generate-metadata.c $(SHA256_DIR)/sha256.c
+METADATA_HDR := $(RIOT_INCLUDE)/hashes/sha256.h
+
+CFLAGS += -g -O3 -Wall -Wextra -pedantic -std=c99
+
+all: bin/generate-metadata
+
+bin/:
+	mkdir -p bin
+
+bin/generate-metadata: bin/ $(METADATA_HDR) $(METADATA_SRC)
```

Comment:
`bin/` should be a pre-requirement instead of a requirement. A pre-requirement only needs to exist, but a newer timestamp does not force a rebuild. Creating `bin/generate-metadata` will touch `bin/`, which will tell `make` to recreate `bin/generate-metadata` in its next invocation. Pre-requirements are placed after the requirements, separated by a pipe.